### ISSUE TITLE
Publish Zettlr on Flathub

### ISSUE THREAD

**Issue by ploum** [Zettlr/Zettlr#1084]
Flatpak seems to be the new default way to distribute end-user software on many linux distribution. Flathub is the most popular repository of Flatpak.

The work involves creating a manifest and then having it on a Github repository created by Flathub ( see https://github.com/flathub/flathub/wiki/App-Submission )

The good point : this work has to be done only one. As long as the way to build the app doesn't change, new versions will be, AFAIK, automatically pushed to Flathub.

The bad point : creating this manifest is far from trivial.

This ticket is thus a good way for a Linux user, preferentially familiar with flatpak, to contribute.

**Comment by nathanlesage** [comment]
> The bad point : creating this manifest is far from trivial.

Hehehe, have you had a look at a CSL-file? :D

**Comment by Thatoo** [comment]
There is already Appimage package of Zettlr (Big +1 to the team for providing .deb and .appimage) that can be installed on all linux distribution. I find flatpak being not as handy as appimage and also heavier.
However, diversity is the best so if fatpak is wanted...

**Comment by ploum** [comment]
AFAIK, appimage doesn't provide repositories and automatic upgrade like Snap and Flatpak. From my observations, there's a move to either Snap or Flatpak, the former being Canonical owned and centralised while the later is more distributed, with Flathub being only one repository (but probably the most popular.

**Comment by Thatoo** [comment]
No, no repository with appimage (could exist though). The WEB is the repository :-)
Automatic update is possible. People who are making the appimage can include this option either with the help of an external tool https://docs.appimage.org/packaging-guide/optional/updates.html#making-appimages-updateable-via-external-tools , usually with such tool https://github.com/AppImage/AppImageUpdate or making the app able to update itself https://docs.appimage.org/packaging-guide/optional/updates.html#making-appimages-self-updateable .

**Comment by BeatLink** [comment]
I've had previous experience writing an app for flathub. I'd love to attempt this!

**Comment by nathanlesage** [comment]
@BeatLink Awesome, that would be great! Just notify me once you did it so that I can add the info on the download page to direct users who prefer FlatHub to that!

**Comment by BeatLink** [comment]
Sure thing@

**Comment by Lunarequest** [comment]
I made a working manifest here https://github.com/Lunarequest/Zettlr-flatpak could I publish this?

**Comment by nathanlesage** [comment]
@Lunarequest Definitely! Thank you very much for this work!

One thing though: According to the Flathub manual, they will run this script to build a version whenever something pushes a commit to the repository, right?

If so, what I could offer to do on my side here would be to implement an action that would just update the information every time we publish a new release. For that, the GitHub Actions user on the Zettlr main repository would just need write access to the repository Flathub creates. This way, all Flathub-users would immediately be able to update just like everyone else. What do you think?

**Comment by Lunarequest** [comment]
I haven't set it up but using f-e-d-c(flatpak external data checker) the flatpak will more or less maintain itself

**Comment by nathanlesage** [comment]
Mh, this flatpak data checker seems to be mainly concerned with third-party data, right? I was talking about the fact that you used hard-coded version numbers in your manifest (currently 2.1.1) and I was thinking to just update those references automatically.

**Comment by Lunarequest** [comment]
> Mh, this flatpak data checker seems to be mainly concerned with third-party data, right? I was talking about the fact that you used hard-coded version numbers in your manifest (currently 2.1.1) and I was thinking to just update those references automatically.

f-e-d-c does that

**Comment by nathanlesage** [comment]
Alright, if that goes smooth, definitely! Go forward :) If I can help with anything, just say so.

**Comment by jaykayes** [comment]
Many thanks for this! For what its worth, I just installed Zettlr with this flatpak manifest with absolutely no issues!

I put high value on the sandbox feature of Flatpaks, and appreciate being able to restrict permissions to the app. Not that I don't trust the developers of Zettlr (quite the opposite), but I feel like Electron could be pretty big attack surface. I've disabled network permissions, pulseaudio, and restricted Zettlr to some whitelisted folders (xdg-documents, and a couple of other specific note folders).

As far as I can tell, Zettlr only uses network to check for updates. Would it be reasonable to build flatpak package without network permissions, in the spirit of principle of least authority? This would disable the update check, but I would argue that for an installation that is managed by flatpak this is not necessary, since Linux distributions seem to default to updating flatpaks automatically.

Unless I'm missing something, pulseaudio permissions are also not needed?

As much as I'd love for filesystem permissions to be restricted as well, I understand that it is not at all reasonable at the moment.

P.S.: I love the naming of the main branch as 'mistress' :D

**Comment by jaykayes** [comment]
As a followup to the previous, I did some testing with my Flatpak installation of Zettlr. For the most part, exporting does not work for me. The behavior in my notes below the line seems to be the same regardless of filesystem permissions being "all of $HOME" or with my restrictions. @Lunarequest does exporting work for you with the Flatpak, in all cases or more than for me at least?

It might be that when I select "Temporary directory" the export itself is succeeding, but the file ends up in an inaccessible place or some form of `/dev/null`. I couldn't find the files inside the sandbox with a quick browse.

Though unfortunate of course, this isn't a showstopper for me. I'm mainly here for the note taking functionality.

Notes from trialling exports.
* * *

- **HTML**: Fails with `An error occurred on export: Could not fetch https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js`. Obviously due to network access being disabled. This is exactly why I want to control permissions, Zettlr is downloading random code from somewhere on the internet!

- **OpenDocument Text**: Works with "Current directory", fails silently on "Temporary directory"
- **Microsoft Word**: Works with "Current directory", fails silently on "Temporary directory"
- **LaTeX Source**: Works with "Current directory", fails silently on "Temporary directory"
- **Orgmode**: Works with "Current directory", fails silently on "Temporary directory"
- **Plain text**: Works with "Current directory", fails silently on "Temporary directory"
- **reStructured Text**: Works with "Current directory", fails silently on "Temporary directory"
- **Markdown**: Works with "Current directory", fails silently on "Temporary directory"
- **Rich text**: Works with "Current directory", fails silently on "Temporary directory"
- **inDesign language**: Works with "Current directory", fails silently on "Temporary directory"

- **PDF Document**: 
For 'Current directory':
`ERR_FILE_NOT_FOUND (-6) loading 'file://<ENTIRE/PATH_TO_FILE/WITH%20SPACES/file%20name.html>'`
For "Temporary directory":
`ERR_FILE_NOT_FOUND (-6) loading 'file:///tmp/<FILE‰20NAME>.html'`

- **PDF LaTeX**: Fails for both current and temporary directories due to not finding `xetex`. No surprise here, since Zettlr can't access software outside the sandbox. The extremely inelegant solution here, though I don't know if it is possible, would be to bundle XeTex inside the flatpak sandbox! 
 
- **reveal.JS presentation**: Works with "Current directory", fails silently on "Temporary directory"
- **Textbundle**: Works with "Current directory", fails silently on "Temporary directory"
- **Textpack**: Works with "Current directory", fails silently on "Temporary directory"

**Comment by nathanlesage** [comment]
I see that you are mistrusting of software, and I've built Zettlr so that it will work without network access.

> Obviously due to network access being disabled. This is exactly why I want to control permissions, Zettlr is downloading random code from somewhere on the internet!

I think you're being overly paranoid here, to a degree I don't fully follow your reasoning.

It's not random code. What you see is a JavaScript file that enables you to display math equations within HTML documents. That is actually being downloaded by [Pandoc](https://github.com/jgm/pandoc/), which we use for exporting purposes. You can also configure Pandoc in the assets manager to not download anything. But convenience-wise, this is perfectly fine to being the default. Have a look at [Pandoc's documentation](https://pandoc.org/MANUAL.html) to see what you can do.

So, fully disabling network access will not break Zettlr since Zettlr should work fully offline. However, it does take some convenience from you, which is why I would like to emphasize that disabling network access *by default* is not the way we should go here. Especially since users who are not technically affine will probably think that Zettlr is broken and not use it anymore. People who know what disabling network access implies will be able to quickly do so with no problems. But we must make sure Zettlr is useable by users who have little to no technical knowledge, that's one of the central tenets of the app.

Then, with regard to exports failing and PDF not working: Zettlr needs permissions to run the LaTeX binary, and since you didn't give Zettlr those permissions, it shouldn't surprise that it couldn't run LaTeX which is installed outside of the flatpak container. Additionally, the LaTeX binary (i.e. xetex) needs to have its own set of access rights since it will also need to read and write some files.

I do agree that Electron is quite the large attack surface, but both the Electron developers as well as me try their best to reduce that. I can reassure you that there is little a malicious actor can do to use Zettlr to take over your computer. We have had our share of CVEs in the beginning of 2021, and believe me, I did not take those carelessly! Furthermore, wouldn't you agree that, if Zettlr was really problematic and was really downloading "random code", someone else might've said something already?

***

**TL;DR**: When you want to distribute the app via flatpak, **do not** disable network access by default. Also make sure to give it the correct permissions to run external programs and read/write access to the temp folder. Always remember: Make it so that Zettlr runs without any additional hassle for less technically versed people. Those who *are* technically versed will be able to withdraw permissions as they see fit.

**Comment by jaykayes** [comment]
Sorry, using the word "random" gave my earier message a very wrong tone. I of course understood from the context what the purpose of the downloaded code would be. I'd like to fully retract my use of the word "random", that was completely inappropriate.

I was taken by surprise that exporting a document involves executing downloaded code. Also, I did not realize that it was pandoc downloading code on demand, and did not even know it had a mechanism to do so. I understand why this is being done, but it does worry me a bit that it is becoming so normalized for applications to download and execute code on the fly. It is of course being done with the best of intent and with the user in mind. I would assume that there is some form of authentication happening on the downloaded code to mitigate against malicious code, but I don't know how robust and dependable that check is. Supply chain attacks are a thing, and I am somewhat worried that this can be a vector for such.

Thanks for the link, I'll give that a browse and consider how I use and set up pandoc in light of that. 

> I think you're being overly paranoid here

I somewhat think I'm being overly paranoid as well! I'm trying to prioritize security as a matter of principle, which does lead to me imposing some irrational feeling restrictions on myself. On the other hand, all of my life is on this computer - from the most mundane everyday photos to sensitive things like tax records, accounting, scans of passport and similar private documents. All my eggs are in one basket so to speak, so there is a lot at stake if a breach was to happen.

I don't distrust Zettlr at all. I have complete faith in you that Zettlr is not doing malicious and that you are doing everything possible to prevent that from happening. Same goes for the developers of Electron and other commonly used frameworks. But more complexity leads to more risk and possibility of unexpected things, hence my paranoia and hesitance with offering unrestricted access to apps.

> Then, with regard to exports failing and PDF not working: Zettlr needs permissions to run the LaTeX binary, and since you didn't give Zettlr those permissions, it shouldn't surprise that it couldn't run LaTeX which is installed outside of the flatpak container.

I'm happy to make this compromise and build pdfs manually, though admittedly the direct LaTeX export is a big selling point of Zettlr! If I understand the intricacies flatpak correctly (and there is a big chance I don't), even with the most lax possible restrictions on the sandbox, a flatpaked Zettlr won't be able to execute commands outside the sandbox. So probably LaTeX exports will be broken on all flatpak installations?

The same might apply with exporting to temporary folders, without some flatpak specific logic.

> Always remember: Make it so that Zettlr runs without any additional hassle for less technically versed people. Those who are technically versed will be able to withdraw permissions as they see fit.

I 100% agree with this. I would not want the user experience I created for myself for anyone else. I understand what is happening and why, and am happy to deal with it. And can undo the changes if necessary. Most users are not in that position. While writing my original message, I did not realize what would break by disabling network access. I value it a lot that Zettlr is being developed to function without constant network access as a hard dependency. The small loss in functionality I experience is unfortunate, but I'll accept it.

Hopefully in the future the frameworks we use will advance so that addressing security issues becomes more transparent to both developer and user. The user experience with sandboxing is definately not polished yet, but I personally do think that flatpak are headed in the right direction.

Anyway this discussion is going quite off topic here...

I appreciate there being a flatpak build. If I can help with this, for example with testing, I'd be happy to participate.

**Comment by Lunarequest** [comment]
I realised it was missing LaTeX binaries. I'll work over the weekend and add em into the flatpak

**Comment by nathanlesage** [comment]
> but it does worry me a bit that it is becoming so normalized for applications to download and execute code on the fly.

Well, that's where the world is headed to. In general I'd also prefer having all of that on a computer, but since all of industry is kind of working its way to put computers as interfaces rather than stand-alone devices it's hard to swim against the stream here.

(Plus, I think more networking is actually good; we should't head back into the siloed nature of early 90's computers, but rather change the political economy of the internet, but that's waaaaay off topic for here.)

> Edit: What path are the "temporary directory" exports being written to? I might be interested in exploring the sandboxing settings to see if I can make this feature work.

Zettlr simply calls a method that is expected to return some temporary path where Zettlr can write stuff to. It doesn't care where it is, but Zettlr uses this mechanism because this way you don't have so much clutter lying around. Zettlr needs to create and read numerous files during operation, and pushing them into this auto-managed, auto-cleaned directory is much easier to handle than putting everything in working directories. So in principle you could give Zettlr some kind of trashy folder somewhere on your computer if you want to. But I don't know which mechanism Electron uses to find the temporary directory, I simply call `app.getPath('temp')`

**Comment by jaykayes** [comment]
> I simply call `app.getPath('temp')`

As you should of course, better to rely on the framework doing the right thing.

It works out to `/tmp` for me, so at least for me allowing filesystem access to `/tmp` gets the exports to work. Just needs `--filesystem=/tmp` added to the manifest to work.

**Comment by Lunarequest** [comment]
> > I simply call `app.getPath('temp')`
> 
> As you should of course, better to rely on the framework doing the right thing.
> 
> It works out to `/tmp` for me, so at least for me allowing filesystem access to `/tmp` gets the exports to work. Just needs `--filesystem=/tmp` added to the manifest.

I'll add it to the flatpak

**Comment by Lunarequest** [comment]
> Then, with regard to exports failing and PDF not working: Zettlr needs permissions to run the LaTeX binary, and since you didn't give Zettlr those permissions, it shouldn't surprise that it couldn't run LaTeX which is installed outside of the flatpak container. Additionally, the LaTeX binary (i.e. xetex) needs to have its own set of access rights since it will also need to read and write some files.
@nathanlesage which flavor of LaTeX does zettlr use?

**Comment by nathanlesage** [comment]
@Lunarequest It defaults to the `xelatex` binary, but basically anything works. Here's the corresponding section in Pandoc's manual: https://pandoc.org/MANUAL.html#creating-a-pdf

Feel free to also have a look at [the `--pdf-engine` option](https://pandoc.org/MANUAL.html#option--pdf-engine).

I think the most reasonable choice would be to add `xetex` since then it works out of the box.

**Comment by Lunarequest** [comment]
> @Lunarequest It defaults to the `xelatex` binary, but basically anything works. Here's the corresponding section in Pandoc's manual: https://pandoc.org/MANUAL.html#creating-a-pdf
> 
> Feel free to also have a look at [the `--pdf-engine` option](https://pandoc.org/MANUAL.html#option--pdf-engine).
> 
> I think the most reasonable choice would be to add `xetex` since then it works out of the box.

I looked into it and I need to ship texlive

**Comment by nathanlesage** [comment]
Sounds good. Just out of curiosity: How large is the flatpak now …?

**Comment by Lunarequest** [comment]
> Sounds good. Just out of curiosity: How large is the flatpak now …?

Without texlive its about the same as the appimage. I haven't built with texlive yet.

**Comment by barthalion** [comment]
@Lunarequest Can you submit it to Flathub? We will see it from here what to do about LaTeX.

**Comment by Lunarequest** [comment]
> @Lunarequest Can you submit it to Flathub? We will see it from here what to do about LaTeX.

sure

**Comment by oskarkook** [comment]
Exposing `/tmp` is not good behaviour I believe. I have overriden the `TMPDIR` environment variable to `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"` per [the instructions here](https://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access) in other manifests. Not sure if Electron respects it though.

**Comment by Lunarequest** [comment]
any ideas what to do about latex, shipping texlive is just not feasible. the size of the flatpak increase to almost 5Gb

**Comment by jaykayes** [comment]
> any ideas what to do about latex, shipping texlive is just not feasible. the size of the flatpak increase to almost 5Gb

Maybe something can be done with `flatpak-spawn`? Might require entering a custom command into Zettlr, or using some kind of shim script in the sandbox.

**Comment by Lunarequest** [comment]
> > any ideas what to do about latex, shipping texlive is just not feasible. the size of the flatpak increase to almost 5Gb
> 
> Maybe something can be done with `flatpak-spawn`? Might require entering a custom command into Zettlr, or using some kind of shim script in the sandbox.

massive security whole I'd like to avoid. is there a super slim version of LaTeX?

**Comment by oskarkook** [comment]
There seems to be a texlive [extension](https://github.com/flatpak/flatpak/wiki/Extensions) available on Flathub: https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive

The TeXstudio application seems to make use of it: https://github.com/flathub/org.texstudio.TeXstudio/blob/master/org.texstudio.TeXstudio.yaml#L5-L10

Worth a try? Users who need LaTeX could then install the extension.

**Comment by oskarkook** [comment]
I opened a PR for this: https://github.com/Lunarequest/flathub/pull/1

From my testing, it seems to work. It is a 4.8GB download, but it is optional. You need to manually run:
```
flatpak install flathub org.freedesktop.Sdk.Extension.texlive
```

**Comment by nathanlesage** [comment]
That sounds like a reasonable approach and mirrors how other Zettlr users do it: Install Zettlr and have the export "PDF Document" available, and optionally install some version of TeX if they like to also use "PDF (LaTeX)"

**Comment by Lunarequest** [comment]
Sounds good to me. I'll poke around and get a patch out for this functionality

**Comment by Lunarequest** [comment]
I made a rough patch, I'm not sure how to add new strings to the translations could someone point me in that direction.

```patch
diff --git a/source/main/modules/export/pdf-exporter.ts b/source/main/modules/export/pdf-exporter.ts
index 9ed268bd..46fa11ad 100644
--- a/source/main/modules/export/pdf-exporter.ts
+++ b/source/main/modules/export/pdf-exporter.ts
@@ -44,7 +44,12 @@ export const plugin: ExporterPlugin = {
     try {
       await commandExists('pandoc')
     } catch (err) {
-      throw new Error(trans('system.error.no_pandoc_message'))
+      try {
+        await fs.access("/.flatpak-info")
+        throw new Error(trans("system.error.no_texlive_extension"))
+      } catch {
+        throw new Error(trans('system.error.no_pandoc_message'))
+      }
     }
 
     // First file determines the name of the output path, EXCEPT a title is
```

**Comment by nathanlesage** [comment]
@Lunarequest That is unnecessary. We defer the check whether `xetex` is available to Pandoc itself, which will print out a meaningful error message if the user attempts to export to PDF via TeX and doesn't have it available. Rather, we need to add a corresponding section to the docs.

**Comment by Lunarequest** [comment]
Okay, that makes sense.  Would you mind adding it?

**Comment by nathanlesage** [comment]
Yes, if you could give me a quick hint as to what Flathub users need to do, I can do that!

**Comment by Lunarequest** [comment]
> Yes, if you could give me a quick hint as to what Flathub users need to do, I can do that!
> 
> I.e.: They don't need to install a TeX distro, but rather execute this command here, right?
> 
> ```
> flatpak install flathub org.freedesktop.Sdk.Extension.texlive//21.08
> ```

yes and no, they are installing a tex distro(TexLive) in a place that the flatpak container can access

**Comment by oskarkook** [comment]
> Yes, if you could give me a quick hint as to what Flathub users need to do, I can do that!
> 
> I.e.: They don't need to install a TeX distro, but rather execute this command here, right?
> 
> ```
> flatpak install flathub org.freedesktop.Sdk.Extension.texlive//21.08
> ```

Yes, that is the command. The Flatpak will use it automatically if it has been installed.

You can also leave the `flathub` part out:
```
flatpak install org.freedesktop.Sdk.Extension.texlive//21.08
```

The user needs to have the [flathub](https://flathub.org/home) repo set up, although if they've managed to install Zettlr as a Flatpak, they probably have already done so. Note that `21.08` can change in the future.

**Comment by Lunarequest** [comment]
> > Yes, if you could give me a quick hint as to what Flathub users need to do, I can do that!
> > I.e.: They don't need to install a TeX distro, but rather execute this command here, right?
> > ```
> > flatpak install flathub org.freedesktop.Sdk.Extension.texlive//21.08
> > ```
> 
> Yes, that is the command. The Flatpak will use it automatically if it has been installed.
> 
> You can also leave the `flathub` part out:
> 
> ```
> flatpak install org.freedesktop.Sdk.Extension.texlive//21.08
> ```
> 
> The user needs to have the [flathub](https://flathub.org/home) repo set up, although if they've managed to install Zettlr as a Flatpak, they probably have already done so. Note that `21.08` can change in the future.

it's recommended by the flatpak folks to include the remote `flathub` in this case. I'd rather keep it

**Comment by nathanlesage** [comment]
Alright, so just so that I can understand it: If there is already a TeXLive installed on the machine, that extension will just make that available to Zettlr, and if it is not already installed, it will install it? Or do users then potentially have TWO TeXLive distros on their computer, one regular install and one via the extension?

**Comment by Lunarequest** [comment]
> Alright, so just so that I can understand it: If there is already a TeXLive installed on the machine, that extension will just make that available to Zettlr, and if it is not already installed, it will install it? Or do users then potentially have TWO TeXLive distros on their computer, one regular install and one via the extension?

More or less

**Comment by oskarkook** [comment]
The Flatpak will not use packages that have been installed through the native package manager (dnf, apt, pacman etc). Instead, the extension contains all the binaries and assets of TeXLive. This is shared between all the other Flatpak applications that make use of the extension.

**Comment by oskarkook** [comment]
Something that I'm wondering about is the automatic update check in Zettlr. Flatpaks have their own update mechanism and if the user has auto-updates enabled through Gnome Software (turned on by default when you use a Gnome-based distro), they will get notified of updates through there. The Firefox Flatpak, which is distributed by Mozilla themselves, for example has the update check turned off.

Would it make sense to turn this off by default and is that even possible?

**Comment by nathanlesage** [comment]
> Would it make sense to turn this off by default and is that even possible?

Yes, there's even a PR for that open. So we'll gradually remove the update notification check for installations that have something to do with package managers and are thus updated differently (Homebrew, Flatpak, Arch, etc.)

**Comment by barthalion** [comment]
I left some minor comments on the Flathub's PR. The main "functional" issue is that the fact that user should install `org.freedesktop.Sdk.Extension.texlive//21.08` has to be documented somewhere – any opinion if it belongs to Zettlr documentation, or rather we should place it in the app description on Flathub?

**Comment by Altonss** [comment]
> The main "functional" issue is that the fact that user should install `org.freedesktop.Sdk.Extension.texlive//21.08`

Can't the extension be installed as a bundle when zettlr is installed?

**Comment by Lunarequest** [comment]
> > The main "functional" issue is that the fact that user should install `org.freedesktop.Sdk.Extension.texlive//21.08`
> 
> Can't the extension be installed as a bundle when zettlr is installed?

the whole point of using the bundle is that TexLive is massive > 4GB hence having it installed as a bundle isn't a good idea

**Comment by Lunarequest** [comment]
> I left some minor comments on the Flathub's PR. The main "functional" issue is that the fact that user should install `org.freedesktop.Sdk.Extension.texlive//21.08` has to be documented somewhere – any opinion if it belongs to Zettlr documentation, or rather we should place it in the app description on Flathub?

I think both is a good idea

**Comment by Altonss** [comment]
> the whole point of using the bundle is that TexLive is massive > 4GB hence having it installed as a bundle isn't a good idea

Ah ok I understand, you're right! I have texlive installed from my distro's repo :)

**Comment by oskarkook** [comment]
> I left some minor comments on the Flathub's PR. The main "functional" issue is that the fact that user should install `org.freedesktop.Sdk.Extension.texlive//21.08` has to be documented somewhere – any opinion if it belongs to Zettlr documentation, or rather we should place it in the app description on Flathub?

An additional option is to document it in the app description and link to the Flatpak app description (if possible) from Zettlr docs. This way the documentation is maintained and kept up to date with the Flatpak, but users can still find it from Zettlr docs.

**Comment by Lunarequest** [comment]
sorry for note being around for a bit, I'm dealing with some legal issues. I've done the step you suggested on my end regarding texlive, All that's left is for it to be merged into flathub which shouldn't take long

**Comment by Lunarequest** [comment]
its been merged! We can close thisn

**Comment by nathanlesage** [comment]
Wonderful! I'll make sure to add it to the website first and then close this issue.

**Comment by nathanlesage** [comment]
Added to the website. Closing issue

**Comment by petesimonic** [comment]
Congratulations and thank you! Where is it appropriate to file a proposal so that Zettlr could follow symlinks? Users have their notes in different places, and I can explain them why Zettlr limited to the Docs folder (due to Flatpak security model), but can't explain why the links to the needed folders doesn't work.

**Comment by Lunarequest** [comment]
If the symlinks lead out of docs the secuirty model blocks them. You may want to consider using flatseal to give access to `home`

**Comment by petesimonic** [comment]
I worry that people won't bother to go that deep, they tend to use more obvious workarounds.

Is it Flatpak limitation or deliberate decision?

**Comment by Lunarequest** [comment]
Yeah its a deliberate decision to limit it to documents. There isn't really a need to go beyond it

**Comment by petesimonic** [comment]
I meant the symlinks. 

> There isn't really a need to go beyond it

I would tend to agree if we were talking about _workspaces_. But Zettlr is great as a text editor with wide area of applications. Let's see how well would it play-out for general public.

**Comment by Etienne-TR** [comment]
I just install Zettlr from flathub and I can access only Documents directory. The post is a bit long and I don't well understand english. Is there a way to have access to the hole user home directory ?
Ideally, I think this access should be by default that would be nice for flatpak.

**Comment by Lunarequest** [comment]
Tldr use flatseal

**Comment by Etienne-TR** [comment]
Thanks for tip, flatseal works well and easy !
